src/main/java/ch/wisv/events/sales/controller/stats/SalesStatsEventController.java

```diff
@@ -0,0 +1,87 @@
+package ch.wisv.events.sales.controller.stats;
+
+import ch.wisv.events.core.exception.normal.EventNotFoundException;
+import ch.wisv.events.core.model.event.Event;
+import ch.wisv.events.core.model.order.Order;
+import ch.wisv.events.core.model.product.Product;
+import ch.wisv.events.core.service.event.EventService;
+import ch.wisv.events.core.service.order.OrderService;
+import ch.wisv.events.sales.service.SalesService;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import java.util.List;
+
+/**
+ * SalesScanEventController.
+ */
+@Controller
+@RequestMapping(value = "/sales/stats/event/{key}")
+@PreAuthorize("hasRole('USER')")
+public class SalesStatsEventController {
+    /** Default return redirect on error. */
+    private static final String ERROR_REDIRECT = "redirect:/sales/stats/";
+
+    /** OrderService. */
+    private final OrderService orderService;
+
+    /**
+     * EventService.
+     */
+    private final EventService eventService;
+
+    /**
+     * SalesService.
+     */
+    private final SalesService salesService;
+
+    /**
+     * SalesScanMainController.
+     *
+     * @param salesService of type SalesService.
+     */
+    public SalesStatsEventController(SalesService salesService, EventService eventService, OrderService orderService) {
+        this.salesService = salesService;
+        this.eventService = eventService;
+        this.orderService = orderService;
+    }
+
+    /**
+     * Ticket index view.
+     *
+     * @param model of type Model
+     * @param key    of type String
+     * @return String
+     */
+    @GetMapping
+    public String index(Model model, @PathVariable String key) {
+        Event event;
+        List<Order> orders;
+        try {
+            event = eventService.getByKey(key);
+            orders = salesService.getAllOrdersByEvent(event);
+        } catch(EventNotFoundException e){
```

Comment:
Maybe catch this using a ExceptionHandler? See here for an example: https://github.com/AreaFiftyLAN/lancie-api/blob/master/src/main/java/ch/wisv/areafiftylan/users/controller/UserRestController.java#L188-L191

Any exception thrown within your controller will use that to decide what to do.

Reply:
Done